thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/alert/commons/AnomalySource.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.thirdeye.alert.commons;
+
+import com.linkedin.thirdeye.datalayer.util.Predicate;
+
+
+public enum AnomalySource {
+  DATASET {
+    @Override
+    public Predicate getPredicate(String predicateValue) {
```

Comment:
This seems domain specific to database fetching. you may want to keep this out of the enum and instead have an enum->predicate map where it actually applies.

Reply:
Do you suggest to have a pure enum, and determine the predicate on the other side? May I know  the reason behind the choice? Thanks for the comment. I am curious about the choice.